### PR TITLE
Use future for py3 compat

### PR DESCRIPTION
**Pull request by loicgasser**
No functional change

This is a simple refactoring so that 

we centralize python 3 compatibility functions
we use the `future` module when we can (doc: http://python-future.org/)

I also added a test for 0 area polygons as it was missing

### CODE REVIEW

File: mapbox_vector_tile/encoder.py [hunk 1]
```diff
@@ -283,17 +268,16 @@ def _handle_attr(self, layer, feature, props):
...
-                    elif (isinstance(v, str if PY3 else unicode)):
+                            val.string_value = unicode(v, 'utf-8')
+                    elif isinstance(v, unicode):
                         val.string_value = v
-                    elif (isinstance(v, int)) or (
-                            isinstance(v, int if PY3 else long)):
+                    elif isinstance(v, int) or isinstance(v, long):
                         val.int_value = v
-                    elif (isinstance(v, float)):
+                    elif isinstance(v, float):
```
Comment (rmarianski) [inline mapbox_vector_tile/encoder.py]: What do you think of spelling the `or isinstance` checks with a tuple instead? For example:

```python
isinstance(v, (str, unicode))
```

Comment (loicgasser) [inline mapbox_vector_tile/encoder.py]: I think it s nice, I honestly didn t know we could do that. Thanks for the tip.

**Comment by loicgasser** [comment]
comment adressed, thanks

**Comment by rmarianski** [comment]
Looks good to me. Thanks for the pr!

**Pull request merged**